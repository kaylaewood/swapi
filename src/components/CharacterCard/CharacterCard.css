.character-card {
  background-color: #0B1E36;
  box-shadow: 15px 15px 15px #A6A6A6;
  color: white;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  margin: 2vw;
  padding: 2vw;
  text-align: center;
}

h4 {
  padding-top: 2vw;
}

.favorite-button {
  font-size: 1vw;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: .5vw;
  width: 15vw;
}
