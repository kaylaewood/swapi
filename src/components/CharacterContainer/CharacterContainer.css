.character-container {
  margin-top: 3vw;
  display: flex;
  overflow: auto;
}
