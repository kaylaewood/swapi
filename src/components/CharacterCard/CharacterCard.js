import React, { Component } from 'react';
import './CharacterCard.css';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

function CharacterCard({ character }) {
  const showFilms = () => {
    let counter=0;
    return character.films.map(film => {
      counter++;
      return <p key={counter}>- {film}</p>
    })
  }

  return (
    <section className='character-card'>
      <h2>{character.name}</h2>
      <h4>SPECIES: <br /> {character.species}</h4>
      <h4>HOMEWORLD: <br /> {character.homeworld[0]}</h4>
      <h4>POPULATION: <br /> {character.homeworld[1]}</h4>
      <h4>FILMS:</h4>
      {showFilms()}
    </section>
  )
}

CharacterCard.propTypes = {
  character: PropTypes.object
}

export default CharacterCard;
