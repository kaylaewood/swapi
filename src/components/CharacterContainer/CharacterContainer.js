import React, { Component } from 'react';
import './CharacterContainer.css';
import CharacterCard from '../CharacterCard/CharacterCard';
import Loader from '../Loader/Loader';
import { getCharacter, getSpecies, getHomeworld, getFilm } from '../apiCalls/apiCalls';

class CharacterContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: true,
      characters: []
    }
  }

  componentDidMount() {
    let characters = [];
    console.log(this.props.characterURLs);
    this.props.characterURLs.map(character => {
      return getCharacter(character)
      .then(data => {
        let name = data.name;
        let speciesInfo = getSpecies(data.species[0]);
        let homeworldInfo = getHomeworld(data.homeworld);
        let films = data.films.map(film => getFilm(film))

        Promise.all([name, speciesInfo, homeworldInfo, ...films])
          .then(data => {
            characters.push({
              name: data[0],
              species: data[1],
              homeworld: data[2],
              films: data.slice(3)
            })
            if (characters.length > 9) {
              this.setState({
                isLoading: false,
                characters: characters
              })
            }
          })
      })
    })
  }

  renderCards = () => {
    let counter = 0;
    return this.state.characters.map(character => {
      counter++;
      return (
        <CharacterCard
          character={character}
          key={counter}
        />
      )
    })
  }

  render() {
    return (
    <section className='character-container'>
      {
        this.state.isLoading ?
        <Loader /> :
        this.renderCards()
      }
    </section>
    );
  }
}

export default CharacterContainer;
