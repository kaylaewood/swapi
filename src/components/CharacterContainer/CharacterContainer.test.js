import React from 'react';
import ReactDOM from 'react-dom';
import CharacterContainer from './CharacterContainer';
import CharacterCard from '../CharacterCard/CharacterCard';

import { shallow } from 'enzyme';

describe('CharacterContainer', () => {
  let characterURLs = ['url1', 'url2']
  let wrapper;
  let character1 = {
    films: ['URL', 'anotherURL'],
    homeworld: 'URL',
    species: 'URL',
    name: 'Billy'
  }
  let character2 = {
    films: ['URL', 'anotherURL'],
    homeworld: 'URL',
    species: 'URL',
    name: 'Bob'
  }

  beforeEach(() => {
    wrapper = shallow(<CharacterContainer
      characterURLs={characterURLs}
    />)

    wrapper.setState({
      isLoading: false,
      characters: [character1, character2]
    })
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should render CharacterCards with correct information', () => {
    expect(wrapper.instance().renderCards().length).toEqual(2);
  })

})
