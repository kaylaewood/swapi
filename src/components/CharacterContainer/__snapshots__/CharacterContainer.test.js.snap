// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterContainer should match the snapshot 1`] = `
<section
  className="character-container"
>
  <CharacterCard
    character={
      Object {
        "films": Array [
          "URL",
          "anotherURL",
        ],
        "homeworld": "URL",
        "name": "Billy",
        "species": "URL",
      }
    }
    key="1"
  />
  <CharacterCard
    character={
      Object {
        "films": Array [
          "URL",
          "anotherURL",
        ],
        "homeworld": "URL",
        "name": "Bob",
        "species": "URL",
      }
    }
    key="2"
  />
</section>
`;
